akka-stream/src/main/scala/akka/stream/impl/QueueSource.scala

```diff
@@ -223,6 +229,16 @@ import akka.stream.stage._
 
       override def fail(ex: Throwable): Unit = callback.invoke(Failure(ex))
 
+      private def handleBufferedOnFailure(ex: Throwable): Unit = {
+        if (buffer != null && buffer.nonEmpty) {
+          onBufferedFailureOrCancel match {
+            case Some(f) =>
+              while (buffer.nonEmpty) f(ex, buffer.dequeue())
+            case None =>
+              log.debug("Queue source failing, [{}] elements in buffer dropped", buffer.used)
```

Comment:
Maybe such events should be fed to dead letters instead to align with other places messages are dropped?

Reply:
do we have that in any other Akka Streams buffers?

Reply:
Nope, can't see we do that anywhere in streams, we probably started doing that with typed. Let's not then.